flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -101,6 +156,13 @@ public boolean rename(Path src, Path dst) throws IOException {
         // The rename is not atomic for ForSt. Some FileSystems e.g. HDFS, OSS does not allow a
         // renaming if the target already exists. So, we delete the target before attempting the
         // rename.
+
+        if (localFileFilter.apply(src.getName())) {
+            Path localSrc = tryBuildLocalPath(src).f1;
+            Path localDst = tryBuildLocalPath(dst).f1;
+            return localFS.rename(localSrc, localDst);
+        }
```

Comment:
The `src` might be a directory, which will fail the `localFileFilter.apply` but still need to be renamed.

Reply:
The directory does not end with `.sst`, hence it can pass the `localFileFilter.apply`